### DISCUSSION TITLE
Custom component with onclick handler

### DISCUSSION THREAD

**Discussion by wackazong** [DioxusLabs/dioxus#2237] [Q&A] [answered]
I have a custom button like this

```rust
#[component]
pub fn IconButton(onclick: Option<EventHandler>) -> Element {
    rsx!(
        a {
            onclick: move |event| if let Some(handler) = onclick{ handler.call(event)},
            Icon {
                ... irrelevant stuff here ...
            }
        }
    )
}
```

Dioxus complains that the `.call()` function expects the unit type `()` as input, not the event. What am I doing wrong here?

**Comment by ealmloff** [comment, marked as answer]
`EventHandler` takes a generic argument for the parameter to the function. If you want to take a function that accepts `MouseEvent` as an argument, you can use `EventHandler<MouseEvent>`

**Reply by wackazong** [reply]
Ok, thanks. That was not easy to spot due to the default generic type of `()`. Thanks for the hint. 

Working version here for future reference:

```rust
#[component]
pub fn IconButton(onclick: Option<EventHandler<MouseEvent>>) -> Element {
    rsx!(
        a {
            onclick: move |event| if let Some(handler) = onclick{ handler.call(event)},
            Icon {
                ... irrelevant stuff here ...
            }
        }
    )
}
```
